src/composable/multiplexer.ts

```diff
@@ -0,0 +1,512 @@
+   */
+  toJSON(): string {
+    const root = this.getOrGenerateTree().root
+
+    // serialize the multiplexer, including the root but excluding the merkle tree.
+    return JSON.stringify({ ...this, root }, (k, v) => {
+      // filter out the merkle tree
+      if (k === 'tree') return undefined
+      if (typeof v === 'object' && v !== null && 'orderType' in v) {
+        const conditionalOrder = v as BaseConditionalOrder<any, any>
+        return {
+          ...conditionalOrder,
+          orderType: conditionalOrder.orderType,
+        }
+      }
+      // We do not do any custom serialization of `BigNumber` in order to preserve it's type.
+      return v
+    })
+  }
+
+  // --- crud methods ---
+
+  /**
+   * Add a conditional order to the merkle tree.
+   * @param order The order to add to the merkle tree.
+   */
+  add<T, P>(order: BaseConditionalOrder<T, P>): void {
+    this.orders[order.id] = order
+    this.reset()
+  }
+
+  /**
+   * Remove a conditional order from the merkle tree.
+   * @param id The id of the `BaseConditionalOrder` to remove from the merkle tree.
+   */
+  remove(id: string): void {
+    delete this.orders[id]
+    this.reset()
+  }
+
+  /**
+   * Update a given conditional order in the merkle tree.
+   * @param id The id of the `BaseConditionalOrder` to update.
+   * @param updater A function that takes the existing `BaseConditionalOrder` and context, returning an updated `BaseConditionalOrder`.
+   */
+  update<T, P>(
+    id: string,
+    updater: (order: BaseConditionalOrder<T, P>, ctx?: string) => BaseConditionalOrder<T, P>
+  ): void {
+    // copy the existing order and update it, given the existing context (if any)
+    const order = updater(this.orders[id], this.ctx)
+    // delete the existing order
+    delete this.orders[id]
+
+    // add the updated order
+    this.orders[order.id] = order
+    this.reset()
+  }
+
+  // --- accessors ---
+
+  /**
+   * Accessor for a given conditional order in the multiplexer.
+   * @param id The `id` of the `BaseConditionalOrder` to retrieve.
+   * @returns A `BaseConditionalOrder` with the given `id`.
+   */
+  getById(id: string): BaseConditionalOrder<any, any> {
+    return this.orders[id]
+  }
+
+  /**
+   * Accessor for a given conditional order in the multiplexer.
+   * @param i The index of the `BaseConditionalOrder` to retrieve.
+   * @returns A `BaseConditionalOrder` at the given index.
+   */
+  getByIndex(i: number): BaseConditionalOrder<any, any> {
+    return this.orders[this.orderIds[i]]
+  }
+
+  /**
+   * Get all the conditional order ids in the multiplexer.
+   */
+  get orderIds(): string[] {
+    return Object.keys(this.orders)
+  }
+
+  get root(): string {
+    return this.getOrGenerateTree().root
+  }
+
+  /**
+   * Retrieve the merkle tree of orders, or generate it if it doesn't exist.
+   *
+   * **CAUTION**: This will overwrite any existing merkle tree. This operation is expensive and should only be done when necessary.
```

Comment:
> This operation is expensive and should only be done when necessary
Give that it only generates when not cached/has been reset, the comment might not fully apply.